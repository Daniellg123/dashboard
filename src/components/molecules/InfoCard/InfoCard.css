.infoCard {
    box-sizing: border-box;

    /* Auto layout */
    display: flex;
    flex-direction: row;
    align-items: flex-start;
    margin: 12px;
    padding: 12px;
    gap: 30%;

    position: static;
    width: 387px;
    height: 42px;

    /* Green/80 Primary */
    border: 1px solid #26A798;
}

.infoSub {
    display: flex;
    flex-direction: row;
}