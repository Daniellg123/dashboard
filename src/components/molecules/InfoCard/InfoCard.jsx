import PropTypes from 'prop-types';
import { Card, Divider, Typography } from "@mui/material";

import "./InfoCard.css";

function InfoCard({ units, users }) {
    return (
        <Card className="infoCard">
            <Typography variant="body4">Child Hierarchy</Typography>
            <div className="infoSub">
                <Typography variant="body3">{units} Units</Typography>
                <Divider variant='middle' />
                <Typography variant="body3">{users} Users</Typography>
            </div>
        </Card>
    );
};

InfoCard.propTypes = {
    /**
     * Units 
     */
    units: PropTypes.number,
    /**
     * Users
     */
    users: PropTypes.number
};

export default InfoCard;