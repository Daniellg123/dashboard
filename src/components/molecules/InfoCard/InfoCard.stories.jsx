import InfoCard from "./InfoCard";

const routeImg = './img/Imageofemployee.png';

export default {
    title: 'Molecules/InfoCard',
    component: InfoCard,
};

const Template = (args) => <InfoCard {...args}/>

export const Primary = Template.bind({});
Primary.args = { title: 'Desarrollo', img: routeImg, units: '130', users: '325', selected: false };