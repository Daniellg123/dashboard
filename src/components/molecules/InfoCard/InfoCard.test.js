import { render } from '@testing-library/react';
import InfoCard from './InfoCard';

const props = {
    units: 12,
    users: 300,
};

it('renders correctly', () => {
    const component = render(
        <InfoCard {...props}/>
    );
    expect(component).toMatchSnapshot();
});

it('Units is equal to prop units', () => {
    const component = render(<InfoCard {...props} units={14}/>)
    component.getByText('14 Units');
});

it('Units is equal to prop users', () => {
    const component = render(<InfoCard {...props} users={14}/>)
    component.getByText('14 Users');
});