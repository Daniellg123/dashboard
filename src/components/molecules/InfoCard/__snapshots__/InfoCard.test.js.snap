// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root infoCard css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body4 css-37jdci-MuiTypography-root"
        >
          Child Hierarchy
        </span>
        <div
          class="infoSub"
        >
          <span
            class="MuiTypography-root MuiTypography-body3 css-37jdci-MuiTypography-root"
          >
            12
             Units
          </span>
          <hr
            class="MuiDivider-root MuiDivider-middle css-ft4jij-MuiDivider-root"
          />
          <span
            class="MuiTypography-root MuiTypography-body3 css-37jdci-MuiTypography-root"
          >
            300
             Users
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root infoCard css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <span
        class="MuiTypography-root MuiTypography-body4 css-37jdci-MuiTypography-root"
      >
        Child Hierarchy
      </span>
      <div
        class="infoSub"
      >
        <span
          class="MuiTypography-root MuiTypography-body3 css-37jdci-MuiTypography-root"
        >
          12
           Units
        </span>
        <hr
          class="MuiDivider-root MuiDivider-middle css-ft4jij-MuiDivider-root"
        />
        <span
          class="MuiTypography-root MuiTypography-body3 css-37jdci-MuiTypography-root"
        >
          300
           Users
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
